functions/index.js

```diff
@@ -1 +1,48 @@
+const fs = require('fs');
+const fetch = require('node-fetch');
+const path = require('path');
+const os = require('os');
+const env = require('dotenv').config();
 const functions = require('firebase-functions');
+const admin = require('firebase-admin');
+const { nanoid } = require('nanoid');
+
+const { Storage } = require('@google-cloud/storage');
+const gcs = new Storage();
+admin.initializeApp();
+
+exports.onUserCreate = functions.firestore.document('/users/{userId}').onCreate(async (snap, context) => {
+  // Grab the current value of what was written to Cloud Firestore.
+  const pictureUrl = snap.data().pictureUrl;
+  const bucket = gcs.bucket(procces.env.FIREBASE_STORAGE);
+  const filename = `${nanoid()}.jpeg`;
+  const filePath = `profile_pics/${filename}`;
```

Comment:
Let's change this a bit.
Create a users folder, which will hold the user images. Should look like this:

```
users/{uid}/profile.jpeg
```